cmake/Corrosion.cmake

```diff
@@ -1086,8 +1085,29 @@ function(corrosion_link_libraries target_name)
             $<TARGET_PROPERTY:${library},LINKER_LANGUAGE>
         )
 
-        corrosion_add_target_local_rustflags(${target_name} "-L$<TARGET_LINKER_FILE_DIR:${library}>")
-        corrosion_add_target_local_rustflags(${target_name} "-l$<TARGET_LINKER_FILE_BASE_NAME:${library}>")
+        if (TARGET ${library})
+            corrosion_add_target_local_rustflags(${target_name}
+                "-L$<TARGET_LINKER_FILE_DIR:${library}>"
+                "-l$<TARGET_LINKER_FILE_BASE_NAME:${library}>"
+            )
+            add_dependencies(_cargo-build_${target_name} ${library})
+            get_property(libs TARGET ${library} PROPERTY INTERFACE_LINK_LIBRARIES)
```

Comment:
`INTERFACE_LINK_LIBRARIES` in CMake is evaluated at build time. This means simply reading this property here would cause us to miss any libraries that are added to `INTERFACE_LINK_LIBRARIES` after `corrosion_link_libraries()`.
We could work around this by making a deferred function call for `TARGET`s, which reads the property at a later point in time.

This is just a nit, it's not required for merging this PR.